### PR TITLE
Adding actions to handle validating urls, generating release notes, and validating schema and samples.

### PR DESCRIPTION
**Pull request by nickkaczmarek**
### What did I add?
- Adds workflow to generate release notes, releases, and validate urls
- Adds workflow to generate git tags and do versioning via PR labels
- Adds question to PR template asking if labels were added
- Adds yaml and samples to editorconfig
- Updates schema description

### The big remaining question(s)?

- should we default to every PR being at least a patch release, or do we want to specify that at label time? 
- So far minor and major are labels that instigate the version number, but I'm not sure if patch should also be as intentional. 
- For instance, is documentation a version update on the workflow schema repo?

## Checklist:
- [x] Did you sign the [Contributor License Agreement](https://cla-assistant.io/wwt/WorkflowSchema)?
- [x] Did you comply with our [styleguide](https://github.com/wwt/WorkflowSchema/blob/main/.editorconfig)?
- [x] Does the CI pipeline pass?
- [ ] Does the schema version need to be changed?
- [ ] Did you update the examples in the schema for the feature you're adding?

### CODE REVIEW

File: .github/workflows/trunk.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+jobs:
+  get_bump_level:
+    if: ${{ github.repository == 'wwt/WorkflowSchema' && github.event.pull_request.merged == true }}
+    runs-on: ubuntu-latest
+    outputs:
+      bump_level: ${{ steps.step1.outputs.bump_level }}
+    steps:
+      - id: step1
+        run: |
+          BUMP_LEVEL=$(gh pr \
```
Comment (Tyler-Keith-Thompson) [inline .github/workflows/trunk.yaml:18]: So `gh` is just available on the ubuntu-latest image?

**Review by Tyler-Keith-Thompson — CHANGES_REQUESTED** [review]

File: .github/workflows/trunk.yaml (hunk 1 above)
Comment (nickkaczmarek) [reply .github/workflows/trunk.yaml:18]: [yes it is](https://docs.github.com/en/actions/advanced-guides/using-github-cli-in-workflows)

File: .github/workflows/trunk.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                        -R $OWNER/$REPO view $PR_NUMBER \
+                        --json labels \
+                        --jq '.labels[]
+                              | select(.name== "minor" or .name == "major")
```
[the remark this answers is not in the document]
Comment (nickkaczmarek) [reply .github/workflows/trunk.yaml:22]: whoops!

File: .github/workflows/trunk.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                              | .name')
+          if [[ ! -z $BUMP_LEVEL && ( $BUMP_LEVEL -eq "minor" || $BUMP_LEVEL -eq "major" ) ]]; then
+              echo "you're good"
```
[the remark this answers is not in the document]
Comment (nickkaczmarek) [reply .github/workflows/trunk.yaml:25]: Yes it is leftover cruft. 😬 I had just noticed it before that last meeting and started working on it. 

And we could have it choose the "highest" level if that's what we wanted to do. I wasn't sure if we wanted to handle that via convention in the PR approval or via the script.

**Review by Tyler-Keith-Thompson — APPROVED** [review]
This looks good. I'm approving it because it works and encouraging you to think about what happens with multiple labels. If the convention is that PRs should contain only one we should probably have the pipeline fail if there are more. If the convention is that PRs can contain multiple and we take the highest then we should code for that. Either way, this is good enough to move forward on for now.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -10,5 +10,6 @@
 - [ ] Does the CI pipeline pass?
 - [ ] Does the schema version need to be changed?
 - [ ] Did you update the examples in the schema for the feature you're adding?
+- [ ] Did you add the appropriate labels to the PR?
```
[the remark this answers is not in the document]
Comment (nickkaczmarek) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: Yeah I was about to add some more details to my existing pr description and thought this would be a good add.

**Comment by nickkaczmarek** [comment]
> This looks good. I'm approving it because it works and encouraging you to think about what happens with multiple labels. If the convention is that PRs should contain only one we should probably have the pipeline fail if there are more. If the convention is that PRs can contain multiple and we take the highest then we should code for that. Either way, this is good enough to move forward on for now.

I think having a pre merge PR pipeline that checks if your labels make sense might be worth looking into.

**Comment by nickkaczmarek** [comment]
I've made it so that the pr checking script can fail and cause subsequent scripts to validate whether it was successful and decide not to run if it wasn't.

I have the ability to not version bump if one isn't provided and having the ability to default to the highest version labeled should be something I can add without too much effort. It looks like `jq` is installed on the actions images so I'll be able to do some nice things without needing to increase our action count. 

I still think we can have a workflow that runs on pr updates that can pre validate that, but the jq stuff should work in the meantime.

**Pull request merged**